docs/using-nativeaot/compiling.md

```diff
@@ -132,6 +132,23 @@ This can be used to import WASI functions that are in other modules, either as t
 Currently NativeAOT-LLVM supports following additional properties
 - `WasmHtmlTemplate`: specifies path to the HTML template within which the WASM application will be embedded. An example of a minimal template can be found in the Emscripten repo: https://github.com/emscripten-core/emscripten/blob/main/src/shell_minimal.html
 
+#### WASM with WASI
+Currently, many things do not work with the WASI publish, notable Exceptions, threads, and any Globalization.  It is also likely to crash when running anything but the simplest programs.  HelloWorld (i.e. `dotnet new console`) does work.
+Set up the project file as above.
+
+Create the wasm with
+```
+dotnet publish -r wasi-wasm -c Debug /p:PlatformTarget=AnyCPU /p:MSBuildEnableWorkloadResolver=false --self-contained```
+`wasmtime` is the only tested runtime.  An example invocation with wasmer:
+```
+wasmer bin\Debug\net8.0\wasi-wasm\publish\console.wasm
+```
+Or with wasmtime passing the required flags:
+```
+wasmtime --wasm-features=threads --wasi-modules=experimental-wasi-threads bin\Debug\net8.0\wasi-wasm\publish\console.wasm
+```
+Note that while we are enabling threads, they do not work attempting to create a thread will most likely crash.
```

Comment:
It crashes, because we don't have an implementation to get the stack trace
```
error: failed to run `bin\Debug\net8.0\wasi-wasm\publish\console.wasm`
╰─▶ 1: RuntimeError: unreachable
           at undefined_stub (console.wasm[32]:0xcb9c)
           at S_P_CoreLib_System_Diagnostics_StackTrace__emscripten_get_callstack (console.wasm[4884]:0x1c88c8)
           at S_P_CoreLib_System_Diagnostics_StackTrace__InitializeForCurrentThread (console.wasm[4648]:0x19440a)
           at S_P_CoreLib_System_Diagnostics_StackTrace___ctor_1 (console.wasm[4400]:0x15fcd0)
           at S_P_CoreLib_System_Exception__DispatchExLLVM (console.wasm[4397]:0x15f844)
           at S_P_CoreLib_System_Runtime_EH__RhpThrowEx (console.wasm[4396]:0x15f6ee)
           at RhpThrowEx (console.wasm[2121]:0x26af1)
           at S_P_CoreLib_System_Threading_Thread__ThrowIfNoThreadStart (console.wasm[4736]:0x1a6098)
           at S_P_CoreLib_System_Threading_Thread__Start_2 (console.wasm[4190]:0x13d367)
           at S_P_CoreLib_System_Threading_Thread__Start_1 (console.wasm[4180]:0x13c1d2)
           at console_Program___Main__ (console.wasm[4177]:0x13b9bd)
           at console__Module___MainMethodWrapper (console.wasm[4509]:0x171657)
           at console__Module___StartupCodeMain (console.wasm[4498]:0x1709c8)
           at main (console.wasm[18208]:0x10e9864)
           at __main_void (console.wasm[18211]:0x10e99ae)
           at _start (console.wasm[48]:0xcc81)
```

Reply:
Kay, so it crashes because it cannot throw PNSE \:).